source/slang/slang-parameter-binding.cpp

```diff
@@ -678,7 +678,7 @@ RefPtr<TypeLayout> getTypeLayoutForGlobalShaderParameter(
     // An "ordinary" global variable is implicitly a uniform
     // shader parameter.
     return createTypeLayout(
-        layoutContext.with(rules->getConstantBufferRules()),
+        layoutContext.with(rules->getConstantBufferRules(context->getTargetRequest())),
```

Comment:
I think that the original plan had been that the "rules" objects would not be parameterized on anything (including the target), and that instead it would be up to the target to construct/return a suitable set of rules objects to describe itself.

As it stands that would get really messy because most of the rules objects are allocated as global-scope values, so I agree with the approach taken here in the short term.

In the long run we should probably have an explicit operation where we take a `TargetRequest` and return a `LayoutRuleFamily*` that implements all its rules, and where the resulting rules objects never need to refer back to the target.